File: request_models/vote.py
from typing import Optional

from pydantic import BaseModel, Field

from enums.enums import VoteType


class VoteCreateRequest(BaseModel):
    user_id: int
    sdg_user_label_id: Optional[int] = Field(default=None, description="The ID of the SDGUserLabel being voted on (optional).")
    annotation_id: Optional[int] = Field(default=None, description="The ID of the Annotation being voted on (optional).")
    vote_type: VoteType
    score: float


    class Config:
        from_attributes = True

File: frontend2/types/vote.ts
import { VoteType } from "./enums";

export interface VoteSchemaBase {
  voteId: number;
  userId: number;
  sdgUserLabelId?: number | null;
  annotationId?: number | null;
  voteType: VoteType;
  score: number;
}

export interface VoteSchemaFull extends VoteSchemaBase {
  createdAt: string;
  updatedAt: string;
}


// Not derived from types
export interface VoteCreateSchema {
  user_id: number;
  sdg_user_label_id?: number | null;
  annotation_id?: number | null;
  vote_type: VoteType;
  score: number;
}
